app/javascript/actionsPanel/actionsPanel.js

```diff
@@ -0,0 +1,321 @@
+import { request } from '@utilities/http';
+
+function addCloseListener() {
+  const button = document.querySelector('.close-actions-panel');
+  button.addEventListener('click', () => {
+    // getting the article show page document because this is called within an iframe
+    const articleDocument = top.document;
+
+    articleDocument
+      .querySelector('.mod-actions-menu')
+      .classList.toggle('showing');
+    articleDocument
+      .querySelector('.mod-actions-menu-btn')
+      .classList.toggle('hidden');
+  });
+}
+
+function initializeHeight() {
+  document.documentElement.style.height = '100%';
+  document.body.style.cssText = 'height: 100%; margin: 0; padding-top: 0;';
+  document.getElementById('page-content').style.cssText =
+    'margin-top: 0 !important; margin-bottom: 0;';
+}
+
+function toggleDropdown(type) {
+  if (type === 'set-experience') {
+    document
+      .querySelector('.set-experience-options')
+      .classList.toggle('hidden');
+  } else if (type === 'adjust-tags') {
+    document.querySelector('.adjust-tags-options').classList.toggle('hidden');
+  }
+}
+
+function correctReactedClasses(category) {
+  const upVote = document.querySelector("[data-category='thumbsup']");
+  const downVote = document.querySelector("[data-category='thumbsdown']");
+  const vomitVote = document.querySelector("[data-category='vomit']");
+
+  if (category === 'thumbsup') {
+    downVote.classList.remove('reacted');
+    vomitVote.classList.remove('reacted');
+  } else {
+    upVote.classList.remove('reacted');
+  }
+}
+
+function addReactionButtonListeners() {
+  const butts = Array.from(
+    document.querySelectorAll('.reaction-button, .reaction-vomit-button'),
+  );
+  /* eslint-disable camelcase */
+  butts.forEach((butt) => {
+    butt.addEventListener('click', (event) => {
+      event.preventDefault();
+      const {
+        reactableType: reactable_type,
+        category,
+        reactableId: reactable_id,
+      } = butt.dataset;
+
+      correctReactedClasses(category);
+      butt.classList.toggle('reacted');
+
+      request('/reactions', {
```

Comment:
If this errors out for whatever reason, a friendly error message should be displayed to the user. The same goes for any other `request`s.

Reply:
I agree, thanks!